packages/http/src/validator/index.ts

```diff
@@ -143,8 +144,21 @@ const validateOutput: ValidatorFn<IHttpOperation, IHttpResponse> = ({ resource,
   );
 };
 
-function getPathParams(path: string, template: string) {
-  return new URI.Template(template).match(path);
+function getPathParams(path: string, template: string, realParams: IHttpPathParam[]) {
+  // replace "-" in template path and query params
+  const cleanedTemplate = template.replace(/(?<=\{.*)(?=.*\})(-)/g, '');
+
+  const pathParams = new URI.Template(cleanedTemplate).match(path);
+
+  // get pathParams back into real format (i.e put "-" back)
+  const finalPathParams = {};
+  for (const realParam of realParams) {
```

Comment:
In general in Prism we do not iterate, we map over objects, so that we do not have side effects or indeterministic functions.

Reply:
So this was building an object from an array, not iterating through an object.

Anyways, changed to use reduce to build the object from the array, although I think it's waaaaay less readable now. 

Done in 779a7f6.

Reply:
I understand it's probably not that readable, but it's now a correct instance of a Monoid, as it should be. 👍 

There's still another step to make this even better but I do not want to overload you with this now :)